tests/IntegrationTests/EventsBuilderTest.php

```diff
@@ -0,0 +1,156 @@
+ */
+class EventsBuilderTest extends IntegrationTestCase
+{
+    /** @test */
+    public function shouldExecuteInteractionCommandOnly(): void
+    {
+        $builder = $this->createMatejInstance()->request()->events();
+        $this->appendInteractionCommands($builder);
+        $response = $builder->send();
+
+        $this->assertResponseCommandStatuses($response, ...$this->generateOkStatuses(1));
+    }
+
+    /** @test */
+    public function shouldExecuteUserMergeCommandOnly(): void
+    {
+        $builder = $this->createMatejInstance()->request()->events();
+        $this->appendUserMergeCommands($builder);
+        $response = $builder->send();
+
+        $this->assertResponseCommandStatuses($response, ...$this->generateOkStatuses(1));
+    }
+
+    /** @test */
+    public function shouldExecuteItemPropertyCommandOnly(): void
+    {
+        $builder = $this->createMatejInstance()->request()->events();
+        $this->appendItemPropertyCommands($builder);
+        $response = $builder->send();
+
+        $this->assertResponseCommandStatuses($response, ...$this->generateOkStatuses(1));
+    }
+
+    /** @test */
+    public function shouldExecuteInteractionAndUserMergeCommandsOnly(): void
+    {
+        $builder = $this->createMatejInstance()->request()->events();
+        $this->appendInteractionCommands($builder);
+        $this->appendUserMergeCommands($builder);
+        $response = $builder->send();
+
+        $this->assertResponseCommandStatuses($response, ...$this->generateOkStatuses(2));
+    }
+
+    /** @test */
+    public function shouldExecuteInteractionAndItemPropertyCommandsOnly(): void
+    {
+        $builder = $this->createMatejInstance()->request()->events();
+        $this->appendInteractionCommands($builder);
+        $this->appendItemPropertyCommands($builder);
+        $response = $builder->send();
+
+        $this->assertResponseCommandStatuses($response, ...$this->generateOkStatuses(2));
+    }
+
+    /** @test */
+    public function shouldExecuteUserMergeAndItemPropertyCommandsOnly(): void
+    {
+        $builder = $this->createMatejInstance()->request()->events();
+        $this->appendUserMergeCommands($builder);
+        $this->appendItemPropertyCommands($builder);
+        $response = $builder->send();
+
+        $this->assertResponseCommandStatuses($response, ...$this->generateOkStatuses(2));
+    }
+
+    /** @test */
+    public function shouldExecuteInteractionAndUserMergeAndItemPropertyCommands(): void
+    {
+        $builder = $this->createMatejInstance()->request()->events();
+        $this->appendInteractionCommands($builder);
+        $this->appendUserMergeCommands($builder);
+        $this->appendItemPropertyCommands($builder);
+        $response = $builder->send();
+
+        $this->assertResponseCommandStatuses($response, ...$this->generateOkStatuses(3));
+    }
+
+    /** @test */
+    public function shouldExecuteOneThousandCommands(): void
```

Comment:
I wouldn't test this at all... This should be part of Matej tests, not part of integration tests for the PHP client - they should check the integration between the systems, not complete behavior of Matej.